### PR TITLE
[SPARK-33086][PYTHON] Add static annotations for pyspark.resource

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

This PR replaces dynamically generated annotations for following modules:

- `pyspark.resource.information`
- `pyspark.resource.profile`
- `pyspark.resource.requests`

### Why are the changes needed?

These modules where not manually annotated in `pyspark-stubs`, but are part of the public API and we should provide more precise annotations.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

MyPy tests:

```
mypy --no-incremental --config python/mypy.ini python/pyspark
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@zero323, let's file a separate JIRA instead of making followups at SPARK-32714 since that ticket basically targets to port as is. It's easier to revert, etc.

**Comment by zero323** [comment]
> @zero323, let's file a separate JIRA instead of making followups at [SPARK-32714](https://issues.apache.org/jira/browse/SPARK-32714) since that ticket basically targets to port as is. It's easier to revert, etc when there's a separate JIRA so it's usually more preferred to create a separate JIRA when possible (at least in this project as far as I can tell).

Sure thing. Please give me a second.

File: python/pyspark/resource/profile.pyi [hunk 1]
```diff
@@ -22,30 +22,39 @@ from pyspark.resource.requests import (  # noqa: F401
...
         self,
-        _java_resource_profile: Optional[Any] = ...,
-        _exec_req: Any = ...,
-        _task_req: Any = ...,
+        _java_resource_profile: Optional[JavaObject] = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self,
+        *,
```
Comment (HyukjinKwon) [inline python/pyspark/resource/profile.pyi:37]: `ResourceProfile.__init__` seems not a keyword-only. Should we maybe change it back?

Comment (HyukjinKwon) [reply python/pyspark/resource/profile.pyi:37]: ohh got it. NVM. the codes assumes it.

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Yeah, this is fine too. LGTM

File: python/pyspark/resource/profile.pyi (hunk 1 above)
Comment (zero323) [reply python/pyspark/resource/profile.pyi]: > ohh got it. NVM. the codes assumes it.

Yeah, it is all about control flow there. But this

```
    @overload
    def __init__(
        self,
        _java_resource_profile: None = ...,
        _exec_req: Dict[str, ExecutorResourceRequest] = ...,
        _task_req: Dict[str, TaskResourceRequest] = ...,
    ) -> None: ...
```

should have the same meaning here.

**Comment by zero323** [comment]
Sorry, @HyukjinKwon [one more](https://github.com/apache/spark/pull/29969/commits/90d89a635f98bbaf434eff063b126dbcc616b931) minor style change.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**